src/common/components/Button/Button.stories.tsx

```diff
@@ -0,0 +1,196 @@
+import type { Meta, StoryObj } from "@storybook/react";
+
+import { Button } from "./Button";
+import { StarLineAltIcon } from "@/common/components/CustomIcon";
+
+const buttonVariants = [
+  "primary",
+  "secondary",
+  "tertiary",
+  "ghost",
+  "success",
+  "danger",
+  "link",
+];
+
+const buttonSizes = ["sm", "md"];
+
+// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
+const meta = {
+  title: "Common/Components/Button",
+  component: Button,
+  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
+  tags: ["autodocs"],
+  // More on argTypes: https://storybook.js.org/docs/api/argtypes
+  argTypes: {
+    variant: {
+      options: buttonVariants,
+      control: {
+        type: "select",
+      },
+    },
+    size: {
+      options: buttonSizes,
+      control: {
+        type: "select",
+      },
+    },
+    fullWidth: {
+      control: {
+        type: "boolean",
+      },
+    },
+    loading: {
+      control: {
+        type: "boolean",
+      },
+    },
+    isResponsive: {
+      control: {
+        type: "boolean",
+      },
+    },
+    rounded: {
+      control: {
+        type: "boolean",
+      },
+    },
+  },
+  args: {
+    variant: "primary",
+    size: "md",
+    children: "primary",
+    as: "a",
+    href: "https://example.com",
+    external: true,
+    loading: false,
+    isResponsive: false,
+    fullWidth: false,
+    rounded: false,
+  },
+} satisfies Meta<typeof Button>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {};
+
+// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
+const VariantsOfButtonTemplate: Story = {
+  render: ({ variant }) => (
```

Comment:
I realise we're not spreading the rest of the props into the button. Could you help me understand the main motivation around generating so many button permutations manually? 

```suggestion
  render: ({ variant, ...props }) => (
```

Reply:
Oh I get it now. Yea that make sense